var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+from spack import *
+
+
+class Faiss(AutotoolsPackage, PythonPackage, CudaPackage):
+    """Faiss is a library for efficient similarity search and clustering of
+       dense vectors.
+
+      Faiss contains algorithms that search in sets of vectors of any size, up
+      to ones that possibly do not fit in RAM. It also contains supporting code
+      for evaluation and parameter tuning. Faiss is written in C++ with
+      complete wrappers for Python/numpy. Some of the most useful algorithms
+      are implemented on the GPU. It is developed by Facebook AI Research.
+    """
+
+    homepage = "https://github.com/facebookresearch/faiss"
+    url      = "https://github.com/facebookresearch/faiss/archive/v1.6.3.tar.gz"
+
+    # putting the creator's name for now
+    #  hopefully, FAISS developers will take over
+    maintainers = ['bhatiaharsh']
+
+    version('1.6.3', sha256='e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542')
+    version('1.5.3', sha256='b24d347b0285d01c2ed663ccc7596cd0ea95071f3dd5ebb573ccfc28f15f043b')
+
+    variant('python', default=False, description='Build Python bindings')
+    variant('tests',  default=False, description='Build Tests')
```

Comment:
hmm.. great point! how do you usually install tests?  you are right, building the tests does not change the binary -- simply another make target.

i think the tests are quite useful since they provide some example runs that validate the use of gpus.. we could always build the tests..

but the tests take quite some time to build.. therefore, i thought it might be better to keep them optional.

Reply:
Do the tests actually end up in the installation prefix? If so, then I would keep the variant. If not, and the tests are only run during `make check`, then you should use `if self.run_tests:` for code or `type='test'` for deps.

Reply:
In that case let's keep the variant.